GithubJobApp/UINetImageView.swift

```diff
@@ -0,0 +1,37 @@
+//
+//  UINetImageView.swift
+//  GithubJobApp
+//
+//  Created by Daniel Harrison on 12/1/16.
+//  Copyright © 2016 Taylor Harrison. All rights reserved.
+//
+
+import Foundation
+import UIKit
+
+class UINetImageView: UIImageView {
+    
+    static let netImageCache = NSCache()
+    
+    func setImgFromUrl(url: String) {
+        if let imgUrl = NSURL(string: url) {
+            let task = NSURLSession.sharedSession().dataTaskWithURL(imgUrl) {(data, response, error) in
+                if error == nil {
+                    dispatch_async(dispatch_get_main_queue()) { () -> Void in
+                        if let data = data, logoImage = UIImage(data: data) {
+                            self.image = logoImage
+                            UINetImageView.netImageCache.setObject(logoImage, forKey: url as AnyObject)
+                        }
+                    }
+                } else {
+                    if let error = error { print("error: \(error) \n") }
+                    dispatch_async(dispatch_get_main_queue()) { () -> Void in
+                        self.image = UIImage(named: "logo_placeholder")
+                    }
+                }
+            }
+            task.resume()
+        }
+    }
+}
+
```

Comment:
Making a subclass of UIImageView to load images on the background is totally valid and works well in this situation. In some cases, you might find yourself using a 3rd party component (or even 1st party) where you can't control the class of the imageView easily.  Instead, you can implement this as a class extension to UIImageView which allows you to add functionality to existing classes.  With a class extension in place, you can then call `setImgFromUrl()`on any UIImageView.  

Go ahead and try implementing this as a class extension.